packages/compass-components/src/components/item-action-controls.tsx

```diff
@@ -388,6 +392,38 @@ export function ItemActionControls<Action extends string>({
     return null;
   }
 
+  // When user wants to show a few actions and collapse the rest into a menu
+  if (collapseAfter > 0) {
+    const visibleActions = actions.slice(0, collapseAfter);
+    const collapsedActions = actions.slice(collapseAfter);
+    return (
+      <div className={actionControlsStyle}>
+        <ItemActionGroup
+          isVisible={isVisible}
+          actions={visibleActions}
+          onAction={onAction}
+          className={cx('item-action-controls', className)}
+          iconClassName={iconClassName}
+          iconStyle={iconStyle}
+          iconSize={iconSize}
+          data-testid={dataTestId}
+        ></ItemActionGroup>
+        <ItemActionMenu
+          isVisible={isVisible}
+          actions={collapsedActions}
+          onAction={onAction}
+          className={cx('item-action-controls', className)}
+          menuClassName={menuClassName}
+          iconClassName={iconClassName}
+          iconStyle={iconStyle}
+          iconSize={iconSize}
+          usePortal={usePortal}
+          data-testid={dataTestId}
+        ></ItemActionMenu>
+      </div>
```

Comment:
looks like we're repeating most of these props 4 times, perhaps we can set define them once and reuse